src/settings.cpp

```diff
@@ -85,6 +85,7 @@ int32_t gen_per_batch {1};
 int64_t n_particles {-1};
 
 int64_t max_particles_in_flight {100000};
```

Comment:
This PR will introduce a good chunk of code that is only hit when more than 100,000 particles are run. I'm not sure if any of the CI tests cover that space at all. E.g., `process_refill_events()` will never be called by a test as it is. Is there a way to make sure the new code gets tested without crazy additions to the test suite?

Reply:
True, should be as simple as reducing the max particles in flight below the generation size for a few of the test cases.

Reply:
So, the question is whether we should simple drop the max particles in flight for some test, or make a new test in some way. I'm not really sure how to proceed on that and think you'll likely have better ideas on this.

I could just copy a simple regression test and call it "event_mode_refilling" and decrease the particles in flight below the particles per generation. What do you think of that?

Reply:
Yeah, appears this code fails if there's any nesting in the geometry. Looking into it now. Guess actually testing it would be worthwhile :smile:

Reply:
> Yeah, appears this code fails if there's any nesting in the geometry. 

Ah interesting -- would not have guessed nested geometry would cause issues! Yes, definitely motivates covering with a test. I don't have strong thoughts on testing strategies (maybe @paulromano has more insight?). One upside I think to just creating a new test(s) (perhaps based off clones of existing ones) is then it can be named something like "event_mode_refill_test" rather than just "pincell", such that if it fails it's more immediately obvious what went wrong.